common/resource.go

```diff
@@ -87,7 +87,10 @@ func (r Resource) ToResource() *schema.Resource {
 	if r.Update != nil {
 		update = func(ctx context.Context, d *schema.ResourceData,
 			m any) diag.Diagnostics {
-			c := m.(*DatabricksClient)
+			c, err := m.(*DatabricksClient).InConfiguredWorkspace(ctx, d)
```

Comment:
it's not enough, you also need to prepend workspace_id to `d.Id()` on a read side and on a write side - e.g. `fmt.Sprintf("%d:%s", d.Get("workspace_id"), d.Id())`. otherwise there's too much risk involved.

Reply:
This field is marked as `ForceNew`, so changes to workspace ID would not trigger update, only delete/recreate. However, we do need to store the old ID somewhere in the state to be able to appropriately move from one workspace to another (as there is no way to get the "old" workspace ID during the delete call).